result/overall.go

```diff
@@ -16,24 +17,28 @@ import (
 // one suffices, but one fails, the whole check might be OK and only the subcheck
 // Warning or Critical.
 type Overall struct {
-	OKs       int
-	Warnings  int
-	Criticals int
-	Unknowns  int
-	Summary   string
-	Outputs   []string // Deprecate this in a future version
+	OKs            int
+	Warnings       int
+	Criticals      int
+	Unknowns       int
+	Summary        string
+	Outputs        []string // Deprecate this in a future version
 	partialResults []PartialResult
 }
 
 type PartialResult struct {
-	State     int
-	Output    string
-	Perfdata  perfdata.PerfdataList
+	State          int
+	Output         string
+	Perfdata       perfdata.PerfdataList
 	partialResults []PartialResult
 }
 
 func (s *PartialResult) String() string {
-	return fmt.Sprintf("[%s] %s|%s", check.StatusText(s.State), s.Output, s.Perfdata.String())
+	if len(s.Perfdata) == 0 {
+		return fmt.Sprintf("[%s] %s", check.StatusText(s.State), s.Output)
+	} else {
```

Comment:
Minor thing, but we don't need that else here. Final return statement is sufficient